src/storage/archive_index.rs

```diff
@@ -21,51 +25,159 @@ impl FileInfo {
     }
 }
 
-#[derive(Deserialize, Serialize)]
-pub(crate) struct Index {
+#[derive(Serialize)]
+struct Index {
     files: HashMap<PathBuf, FileInfo>,
```

Comment:
Another option that might be fast without having to write a custom deserializer is to use zero-copy deserialization (`files: HashMap<&Path, FileInfo>`), that would avoid allocating for each key, but would still require parsing the file-info so would probably end up a little slower still.

Reply:
wouldn't we still have the full `HashMap` in memory? With all `FileInfo`s? 


But just for reading the keys I'll try using references, and see if it makes the process faster.

Reply:
`&[u8]` might be even faster, and is good enough for exact utf-8 lookup (though depending on how `serde-cbor` serializes `Path` it might still validate it's utf-8 while reading it in).

Reply:
`&[u8]` didn't really help.